// ng2-signalr/lib/signalr.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var core_1 = require("@angular/core");
var signalr_1 = require("./signalr");
var SIGNALR_CONFIGURATION = new core_1.OpaqueToken('SIGNALR_CONFIGURATION');
function createSignalr(configuration, zone) {
    var jConnectionFn = getJConnectionFn();
    return new signalr_1.SignalR(configuration, zone, jConnectionFn);
}
exports.createSignalr = createSignalr;
function getJConnectionFn() {
    var jQuery = getJquery();
    var hubConnectionFn = window.jQuery.hubConnection;
    if (hubConnectionFn == null)
        throw new Error('Signalr failed to initialize. Script \'jquery.signalR.js\' is missing. Please make sure to include \'jquery.signalR.js\' script.');
    return hubConnectionFn;
}
function getJquery() {
    var jQuery = window.jQuery;
    if (jQuery == null)
        throw new Error('Signalr failed to initialize. Script \'jquery.js\' is missing. Please make sure to include jquery script.');
    return jQuery;
}
var SignalRModule = SignalRModule_1 = (function () {
    function SignalRModule() {
    }
    SignalRModule.configure = function (configuration) {
        return {
            ngModule: SignalRModule_1,
            providers: [
                {
                    provide: SIGNALR_CONFIGURATION,
                    useValue: configuration
                },
                {
                    provide: signalr_1.SignalR,
                    useFactory: (createSignalr),
                    deps: [SIGNALR_CONFIGURATION, core_1.NgZone]
                }
            ],
        };
    };
    return SignalRModule;
}());
SignalRModule = SignalRModule_1 = __decorate([
    core_1.NgModule({
        providers: [{
                provide: signalr_1.SignalR,
                useValue: signalr_1.SignalR
            }]
    })
], SignalRModule);
exports.SignalRModule = SignalRModule;
var SignalRModule_1;
